docs/guides/ingesters-rolling-updates.md

```diff
@@ -0,0 +1,91 @@
+---
+title: "Ingesters rolling updates"
+linkTitle: "Ingesters rolling updates"
+weight: 102
+slug: ingesters-rolling-updates
+---
+
+Cortex [ingesters](architecture.md#ingester) are semi-stateful.
+A running ingester holds several hours of time series data in memory, before they're flushed to the long-term storage.
+When an ingester shutdowns, because of a rolling update or maintenance, the in-memory data must not be discarded in order to avoid any data loss.
+
+In this document we describe the techniques employed to safely handle rolling updates, based on different setups:
+
+- [Blocks storage](#blocks-storage)
+- [Chunks storage with WAL enabled](#chunks-storage-with-wal-enabled)
+- [Chunks storage with WAL disabled](#chunks-storage-with-wal-disabled-hand-over)
+
+## Blocks storage
+
+The Cortex [blocks storage](../blocks-storage/_index.md) requires ingesters to run with a persistent disk where the TSDB WAL and blocks are stored (eg. a StatefulSet when deployed on Kubernetes).
+
+During a rolling update, the leaving ingester closes the open TSDBs, synchronize the data to disk (`fsync`) and releases the disk resources.
+The new ingester, which is expected to reuse the same disk of the leaving one, will replay the TSDB WAL on startup in order to load back in memory the time series that have not been compacted into a block yet.
+
+_The blocks storage doesn't support the series [hand-over](#chunks-storage-with-wal-disabled-hand-over)._
+
+## Chunks storage
+
+The Cortex chunks storage optionally supports a write-ahead log (WAL).
+The rolling update procedure for a Cortex cluster running the chunks storage depends whether the WAL is enabled or not.
+
+### Chunks storage with WAL enabled
+
+Similarly to the blocks storage, when Cortex is running the chunks storage with WAL enabled, it requires ingesters to run with a persistent disk where the WAL is stored (eg. a StatefulSet when deployed on Kubernetes).
+
+During a rolling update, the leaving ingester closes the WAL, synchronize the data to disk (`fsync`) and releases the disk resources.
+The new ingester, which is expected to reuse the same disk of the leaving one, will replay the WAL on startup in order to load back in memory the time series that have not been flushed to the storage yet.
```

Comment:
It actually loads all data into memory, even flushed samples. WAL doesn't preserve "flushed" flag, so data will be possibly flushed twice.

Reply:
Right. I'm not sure we should mention the "data will be possibly flushed twice" here, given this page is focused on rolling updates. Maybe it's something we should improve in the "Ingesters with WAL" page. WDYT?

In the meanwhile, I've rephrased it inhttps://github.com/cortexproject/cortex/pull/2996/commits/caa7dd453b918cd0674cbc52d999490e3ceebe0b.

Reply:
Nit: In that change you have introduce "checkpoint" term, with no explanation. I think we can be bit more vague and just say that ingester reloads data from WAL.

Reply:
Agree, done.